### PR TITLE
Update DEVGUIDE with using custom FSharp.Core

### PR DESCRIPTION
**Pull request by rbauduin**
## Description

Adding information on how to use a modified FSharp.Core in a project.

### CODE REVIEW

**Comment by rbauduin** [comment]
@dotnet-policy-service agree

File: DEVGUIDE.md [hunk 1]
```diff
@@ -140,6 +140,25 @@ To use your custom build of `Fsc`, add the `DotnetFscCompilerPath` property to y
 </PropertyGroup>
 ```
 
+### Changes in FSharp.Core
+
+The FSharp compiler uses an implicit FSharp.Core. This means that if you introduce changes to FSharp.Core and want to use it in a project, you need to disable the implicit version used by the compiler, and add a reference to your custom FSharp.Core dll. Both are done in the `.fsprof` file of your project.
```
Comment (edgarfgp) [inline DEVGUIDE.md:145]: Typo: `fsprof` -> `fsproj`

**Review by vzarytovskii — APPROVED** [review]
Nice, thanks!

**Review by psfinaki — APPROVED** [review]
There are more ways to do this, but this is a helpful note. Thanks!

**Pull request merged**